packages/flame_test/lib/src/flame_test.dart

```diff
@@ -158,6 +167,12 @@ class GameTester<T extends Game> {
           await _pump(gameWidget, tester);
           await tester.pump();
 
+          if (makeReady && game is FlameGame) {
+            // ignore: invalid_use_of_visible_for_testing_member
+            game.setMounted();
```

Comment:
This looks suspicious to me: if the `game` is being put into a `GameWidget`, which is pumped through a `tester`, then this should be no different than the "normal" use of `FlameGame`. In other words, `setMounted` should not be necessary here...

Which makes me wonder - have we checked that the examples are still working ok?

Reply:
Agree, which examples are you referring to?

Reply:
I just tried running `examples/lib/main.dart` from this branch, and immediately got an error:
```
Assertion failed: file:///flame/packages/flame/lib/src/components/core/component.dart:1024:14
child._parent!.isMounted
```

Reply:
Interesting, so somehow it isn't called in the order I expected then I guess, or not called at all... Any clue what could be wrong? I guess that would solve the sketchy widget tests too

Reply:
Now I'm doing `setMounted` explicitly from the `GameWidget` instead after `onMount`, do you think that is the correct way to do it?

Reply:
Moving the discussion here:
https://github.com/flame-engine/flame/pull/2246/files#r1062345184